SELA-301 Support Mocha without Karma as a test runner.

threehams opened the pull request:
Set up Mocha to run tests without needing to spin up a browser or bundle. This is very close to a drop-in replacement right now.
### Changes
- Assets (scss, svg, png) need to change to use relative paths. `require.extensions` only works for file imports, not for modules, and anything without a relative path is assumed to be a module. This means that anything using a path such as `import ribbonUrl from "assets/img/icons/ribbon.svg"` will pass in the Karma tests, and work in the app, but fail tests using Mocha.

I'm trying to find a way to get this working, but the directory aliasing is done through Webpack, and is not based on any standard.
## Benchmarks
#### Mocha + Karma + Webpack

Startup time: 43 seconds
Time from file save to results: 10-11 seconds
#### Mocha + Node

Startup time: 4 seconds
Time from file save to results: 400-800 ms

threehams committed:
SELA-301 Support Mocha without Karma as a test runner.

christinebrass commented on src/commands/test.js:
```diff
@@ -2,23 +2,38 @@ const karma = require("karma");
+  // override to run tests in Node without Karma/Webpack
+  if (options.mochaOnly) {
+    const helperPath = path.join(__dirname, "..", "lib", "testHelperMocha.js");
+    const testPath = `${CWD}/test/**/*.js`;
```
If we want to be consistent with our karma test helper, we'd use `*.test.js` as opposed to `*.js` at the end here: https://github.com/TrueCar/gluestick/blob/develop/src/lib/testHelper.js#L11

threehams committed:
SELA-301 Support multiple aliases for src + assets.
Filter to only run .test.js files.
Add lots of comments around the wackier bits.

christinebrass commented:
👍
